File: archetype-java/src/main/resources/archetype-resources/src/main/java/cli/HelloCommand.java
#set( $symbol_pound = '#' )
#set( $symbol_dollar = '$' )
#set( $symbol_escape = '\' )
package ${package}.cli;

import org.apache.commons.cli.CommandLine;
import org.apache.commons.cli.Options;
import org.apache.commons.lang.StringUtils;
import org.springframework.shell.core.CommandMarker;
import org.springframework.shell.core.annotation.*;
import org.springframework.stereotype.Component;
import ${package}.resources.Sayer;

import javax.inject.Inject;

@Component
public class HelloCommand implements CommandMarker {

    @Inject
    Sayer sayer;

    @CliAvailabilityIndicator({"hello"})
    public boolean isCommandAvailable() {
        return true;
    }

    @CliCommand(value = "hello", help = "Say hello to the shell")
    public String hello(
            @CliOption(key = "message", mandatory = true, help = "The hello world message")
            String message,

            @CliOption(key = "more", mandatory = false,
                    help = "Have some more to say while saying hello",
                    specifiedDefaultValue="")
            String more)
    {
        StringBuilder hello = new StringBuilder(sayer.hello());
        hello.append(" with this message, "+message);
        if (!StringUtils.isBlank(more)) {
            hello.append(" "+more);
        }
        System.out.println("${symbol_escape}n${symbol_escape}n${symbol_escape}n${symbol_escape}n"+hello+"${symbol_escape}n${symbol_escape}n${symbol_escape}n${symbol_escape}n");
        return null;//hello.toString();
    }
}

File: archetype-scala/src/main/resources/archetype-resources/src/main/scala/cli/HelloCommand.scala
#set( $symbol_pound = '#' )
#set( $symbol_dollar = '$' )
#set( $symbol_escape = '\' )
package ${package}.cli

import javax.inject.{Named, Inject}
import com.netflix.config.scala.{DynamicStringProperty => DString}
import org.springframework.shell.core.CommandMarker
import org.apache.commons.lang.StringUtils
import org.springframework.shell.core.annotation.{CliOption, CliCommand, CliAvailabilityIndicator}
import org.springframework.stereotype.Component

@Component
class HelloCommand @Inject() (@Named("helloText") helloText: DString) extends CommandMarker {

  @CliAvailabilityIndicator(Array("hello"))
  def isCommandAvailable = {
    true
  }

  @CliCommand(value = Array("hello"), help = "Say hello to the shell")
  def hello(
             @CliOption(key = Array("message"), mandatory = true, help = "The hello world message")
             message: String,
             @CliOption(key = Array("more"), mandatory = false, help = "Have some more to say while saying hello",
             specifiedDefaultValue = "") more: String): String =
  {
    val hello: StringBuilder = new StringBuilder(helloText.get())
    hello.append(" with this message, " + message)
    if (!StringUtils.isBlank(more)) {
      hello.append(" " + more)
    }
    System.out.println("\n\n\n\n" + hello + "\n\n\n\n")
    return null
  }
}
